src/plugins/data/common/es_query/es_query/types.ts

```diff
@@ -6,9 +6,29 @@
  * Side Public License, v 1.
  */
 
-import { IFieldType } from '../../index_patterns';
+import type { estypes } from '@elastic/elasticsearch';
+
+export interface IFieldSubType {
+  multi?: { parent: string };
+  nested?: { path: string };
+}
+export interface FieldBase {
```

Comment:
If we're going to rename this, lets call it `IndexPatternFieldBase` - occasionally, there are different representations of fields in kibana and I prefer to be as specific as possible.